shared/src/store/validate-exercise-export.ts

```diff
@@ -0,0 +1,50 @@
+import { plainToInstance } from 'class-transformer';
+import type { ValidationError } from 'class-validator';
+import { validateSync } from 'class-validator';
+import type { ExportImportFile } from '../export-import/file-format';
+import { PartialExport, StateExport } from '../export-import/file-format';
+import type { Constructor } from '../utils';
+import { validateExerciseAction } from './validate-exercise-action';
+import { defaultValidateOptions } from './validation-options';
+
+function fileIsCompleteExport(file: ExportImportFile): file is StateExport {
+    return file.type === 'complete';
+}
```

Comment:
I don't think you need a type guard - TS should get this without one (discriminated union)

Reply:
Unfortunately, it doesn't.